ls1intum/Artemis
# `Assessment`: Add shortcuts to save and submit an assessment

### Checklist
#### General
#### Client

### Motivation and Context


Clicking the `save`, `submit` and `assess next submission` button are frequently used operations for tutors. To increase efficiency, a tutor should be able to save & submit their assessment as well as accessing the next submission using keyboard shortcuts.

### Description

They are displayed as tooltips above the respective button.

Shortcut for **saving** the assessment:
`Control` + `S` (Mac), `Ctrl` + `S` (Windows)

Shortcut for **submitting** and **overriding** the assessment:
`Control` + `Enter` (Mac), `Ctrl`+ `Enter`(Windows)

Shortcut for **assessing next submission**:
`Control` + `Shift` + `ArrowRight` (Mac), `Ctrl`+ `Shift` + `ArrowRight`(Windows)

### Steps for Testing

Prerequisites:
- 1 Tutor
- 1 Student submission of a text/modeling/file-upload/programming exercise(with manual assessment)

1. Navigate to the exercise dashboard
2. Start an assessment
3. Assess something and use the shortcut: `Ctrl` + `S` to save your assessment
4. Make sure your changes are saved
5. Submit your assessment by using the shortcut: `Ctrl`+ `Enter`
6. Make sure your assessment was submitted
7. Override the assessment by using the saving shortcut
8. Go to next submission by using the shortcut: `Ctrl` + `Shift` + `ArrowRight` (You will be informed in case there are no open submissions left)

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots
German version displays `Strg`

<img width="84" alt="Bildschirmfoto 2022-03-24 um 15 48 10" src="https://user-images.githubusercontent.com/73841445/159953421-d3bd9f33-0d54-4348-bfec-78a9f985f94b.png"> <img width="116" alt="Bildschirmfoto 2022-03-24 um 15 52 35" src="https://user-images.githubusercontent.com/73841445/159953432-367f799c-6405-43a5-a155-0c5a0cb4875f.png"> 
<img width="92" alt="Bildschirmfoto 2022-03-24 um 15 48 21" src="https://user-images.githubusercontent.com/73841445/159953427-9c87b2b6-3878-409d-88e2-d811f25f2587.png"> <img width="173" alt="Bildschirmfoto 2022-03-24 um 15 48 52" src="https://user-images.githubusercontent.com/73841445/159953430-61695fed-fe41-4f30-98b9-9e62d7cdad71.png"> <img width="193" alt="Bildschirmfoto 2022-03-24 um 15 48 37" src="https://user-images.githubusercontent.com/73841445/159953428-1d5be3cb-73b2-467b-b221-461147d8e13c.png">
